src/EditorFeatures/CSharp/EventHookup/EventHookupSessionManager.cs

```diff
@@ -104,18 +105,23 @@ internal void CancelAndDismissExistingSessions()
         /// <summary>
         /// If any text is deleted or any non-space text is entered, cancel the session.
         /// </summary>
-        private void TextBuffer_Changed(object sender, TextContentChangedEventArgs e)
+        private  Task TextBuffer_Changed(object sender, TextContentChangedEventArgs e)
         {
-            AssertIsForeground();
-
-            foreach (var change in e.Changes)
+            return Task.Factory.StartNew(() =>
             {
-                if (change.OldText.Length > 0 || change.NewText.Any(c => c != ' '))
+                AssertIsForeground();
+
+                foreach (var change in e.Changes)
                 {
-                    CancelAndDismissExistingSessions();
-                    return;
+                    if (change.OldText.Length > 0 || change.NewText.Any(c => c != ' '))
+                    {
+                        CancelAndDismissExistingSessions();
```

Comment:
You lost the return that meant we'd bail once calling CancelAndDismissExistingSessions().